bitcoinfiles.md

```diff
@@ -81,22 +81,39 @@ The scripts below are designed with the following transaction limits in mind:
 - Max script element size of 520 bytes (consensus limit)
 - Max `scriptSig` size of 1650 bytes per input (standardness limit)
 
-Redeem script shall have the following format:
-`OP_HASH160 OP_SWAP OP_HASH160 OP_CAT OP_2DUP OP_CAT OP_HASH160 <hash> OP_EQUALVERIFY <push1> OP_2DROP OP_CHECKSIGVERIFY OP_DEPTH OP_NOT`
+Unlocking script shall have the following format (adhering to the P2SH specification):
+`<signature> <public key> <push1> <push2> <redeem script>`
 
-Unlocking script shall have the following format:
-`<signature> <pubkey> <push2> <push3>`
+Redeem script shall have the following format:
+`OP_HASH160 OP_SWAP OP_HASH160 OP_CAT OP_2DUP OP_CAT OP_HASH160 <hash> OP_EQUALVERIFY <push3> OP_2DROP OP_CHECKSIGVERIFY OP_DEPTH OP_NOT`
 
-where `hash`: `HASH160(public key || HASH260(push3) || HASH160(push2))`
+where `hash`: `HASH160(public key || HASH160(push2) || HASH160(push1))`
 
 For maximum efficiency, it is suggested that
 
-- `push 1` is of length `480` bytes
-- `push 2` is of length `520` bytes
-- `push 3` is of length `493` bytes (`502` if `signature` is Schnorr-signed)
+- `push 1` is of length `520` bytes
+- `push 2` is of length `493` bytes (`502` if `signature` is Schnorr-signed)
+- `push 3` is of length `480` bytes
+
+The `vout = 0` output of every upload transaction shall be an `OP_RETURN` push.
```

Comment:
Suggest removing ambiguity by adding something along these lines: 

"P2sh outputs shall follow immediately after the `OP_RETURN` at `vout = 1`, they shall be contiguous, and spent in the same order starting at `vin=0`.  For example, `vout=1` would be spent in `vin=0`, `vout=2` spent to `vin=1`, etc."

A pretty picture might also help users conceptualize, but not required.

Reply:
Updated.